File: Anul 2/Semestrul 2/Metode de proiectare si programare/labs/Tema 2-proiect Csharp/proiect/proiect/model/Spectacol.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace proiect.model
{
    public class Spectacol : Entity<long>
    {
        public string artistName { get; set; }
        public DateTime date { get; set; }
        public string place { get; set; }
        public int availableSeats { get; set; }
        public int soldSeats { get; set; }

        public Spectacol(string ArtistName, string Place, int AvailableSeats, int SoldSeats)
        {
            artistName = ArtistName;
            //date = Date;
            place = Place;
            availableSeats = AvailableSeats;
            soldSeats = SoldSeats;
        }
        public string getDateString() 
        { 
            return date.ToString("dd-MM-yyyy HH:mm"); 
        }
        public void setDate(String datee)
        {
            date = DateTime.ParseExact(datee, "dd-MM-yyyy HH:mm",null);
        }
        public override string ToString()
        {
            return artistName + " | " + date + " | " + place + " | " + availableSeats + " | " + soldSeats;
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 1-proiect Java/proiect/src/main/java/com/example/proiect/model/Spectacol.java
package com.example.proiect.model;

import com.example.proiect.utils.Constants;

import java.time.LocalDateTime;

public class Spectacol extends Entity<Long> {
    private String artistName;
    private LocalDateTime date;
    private String place;
    private int availableSeats;
    private int soldSeats;

    public Spectacol(String artistName,String place, int availableSeats, int soldSeats) {
        this.artistName = artistName;
        //this.date = date;
        this.place = place;
        this.availableSeats = availableSeats;
        this.soldSeats = soldSeats;
    }

    public String getArtistName() {
        return artistName;
    }

    public void setArtistName(String artistName) {
        this.artistName = artistName;
    }
    public String getDateString(){
        return date.format(Constants.FORMATTER_MESSAGE);
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = LocalDateTime.parse(date,Constants.FORMATTER_MESSAGE);
    }

    public String getPlace() {
        return place;
    }

    public void setPlace(String place) {
        this.place = place;
    }

    public int getAvailableSeats() {
        return availableSeats;
    }

    public void setAvailableSeats(int availableSeats) {
        this.availableSeats = availableSeats;
    }

    public int getSoldSeats() {
        return soldSeats;
    }

    public void setSoldSeats(int soldSeats) {
        this.soldSeats = soldSeats;
    }
}